### ISSUE TITLE
Komodo Edit 9: clicking 'OK' on 'Current file preferences' clears the 'Places' pane

### ISSUE THREAD

**Issue by brianjcohen** [Komodo/KomodoEdit#328]
Issue did not exist in 8.5.*, can reproduce in: Komodo Edit, version 9.1.0, build 15798, platform linux-x86_64. Built on Sat May 30 05:22:08 2015.

Steps to reproduce:
1. Ensure a file is open. 
2. Ensure there are files and directories visible in the 'Places' pane.
3. Click 'Edit' menu, click 'Current File Preferences'.
4. Click 'OK'.

Expected behavior:
The 'Current file preferences' window should close and everything in the various Komodo Edit panes should remain as they were.

Actual behavior:
The 'Places' pane clears. I have to bring the 'Places' pane back to where it was by double-clicking a folder in the 'Projects' pane.

**Comment by Naatan** [comment]
The pane is reloaded and its blank for a brief moment for me, but it does not stay blank. I assume this is not the same for you? Could you share your error log after reproducing? Help > Troubleshooting > View Log File.

**Comment by brianjcohen** [comment]
Hi Naatan,

You are correct.  Here is my log file contents after reproducing:
```
[2015-06-09 13:43:47,085] [INFO] Startup: Welcome to Komodo Edit 9.1.0 build 15798 (platform linux-x86_64, running on Linux 3.13.0-53-generic version #89-Ubuntu SMP Wed May 20 10:34:39 UTC 2015)
[2015-06-09 13:43:47,085] [INFO] Startup: /usr/bin/python built on Sat May 30 05:22:08 2015
[2015-06-09 13:43:47,755] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2015-06-09 13:43:47,757] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'component' directive. (1) in file:///usr/lib/komodo-edit/mozilla/extensions/scope-files@activestate.com/components/component.manifest:1
[2015-06-09 13:43:47,757] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'contract' directive. (1) in file:///usr/lib/komodo-edit/mozilla/extensions/scope-files@activestate.com/components/component.manifest:2
[2015-06-09 13:43:47,758] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'interfaces' directive. (1) in file:///usr/lib/komodo-edit/mozilla/extensions/scope-files@activestate.com/components/component.manifest:3
[2015-06-09 13:43:47,889] [WARNING] root: ko.logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:48:1

[2015-06-09 13:43:48,486] [WARNING] console-logger: Unknown property 'border-radius-top-left'.  Declaration dropped. (1) in less://commando/skin/commando.less:18
[2015-06-09 13:43:48,487] [WARNING] console-logger: Unknown property 'border-radius-top-right'.  Declaration dropped. (1) in less://commando/skin/commando.less:19
[2015-06-09 13:43:49,229] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///usr/lib/komodo-edit/mozilla/components/koamAddonManager.js:138
[2015-06-09 13:43:58,136] [WARNING] koLanguageServiceBase.indenting: Found non-positive guess of 0 (min 1)
[2015-06-09 13:43:58,250] [WARNING] ko.widgets: Could not restore widget NST-viewbox
[2015-06-09 13:44:01,002] [WARNING] root: ko.findresults was already loaded, re-creating it.

[2015-06-09 13:44:03,231] [WARNING] console-logger: Unexpected value  parsing x1 attribute. (1) in chrome://komodo/skin/images/toolbox/macro.svg:0
[2015-06-09 13:44:03,240] [WARNING] console-logger: Unexpected value  parsing y1 attribute. (1) in chrome://komodo/skin/images/toolbox/macro.svg:0
[2015-06-09 13:44:03,250] [WARNING] console-logger: Unexpected value  parsing x2 attribute. (1) in chrome://komodo/skin/images/toolbox/macro.svg:0
[2015-06-09 13:44:03,258] [WARNING] console-logger: Unexpected value  parsing y2 attribute. (1) in chrome://komodo/skin/images/toolbox/macro.svg:0
[2015-06-09 13:44:03,314] [WARNING] console-logger: Error in parsing value for 'opacity'.  Declaration dropped. (1) in chrome://komodo/skin/images/toolbox/macro.svg:0
[2015-06-09 13:44:03,368] [DEBUG] places_js: waiting for document complete
[2015-06-09 13:44:35,086] [ERROR] console-logger: NS_ERROR_FILE_NOT_FOUND:  (2) in chrome://komodo/content/pref/koFilteredPrefTree.js:681
Traceback (most recent call last):
  File "chrome://komodo/content/pref/koFilteredPrefTree.js", line 681, in 
```

**Comment by Naatan** [comment]
Does this happen on all files? If not, something different about the files that it does happen on? Are you using files on a share/mount perhaps?

**Comment by brianjcohen** [comment]
Yes it happens for me on all files, remote (SFTP) or local. I've never tried it on a share or mount.

**Comment by cgchoffman** [comment]
Is this still any issue?  @brianjcohen, have you had a chance to try Komodo 10 to see if it's still a problem?

**Comment by brianjcohen** [comment]
Yes it is absolutely still a reproducible issue in Komodo 10.

**Comment by Naatan** [comment]
Can you reproduce this using `Help > Troubleshooting > Restart in Safe-mode` ?

Please verify that when restarting in safe-mode Komodo should be using the default UI (all your settings would be temporarily reset).

**Comment by brianjcohen** [comment]
I can confirm that the problem is reproducible in Safe Mode.  I can also confirm that the UI reset to default settings in Safe Mode.

**Comment by Naatan** [comment]
I'm not able to reproduce this for the life of me :\ Tried IDE 10.0.1 and Edit 10.0.1 with various variations of the reproduction steps. I assume opening a project or manually pointing your Places widget at a directory "fixes" this?

**Comment by brianjcohen** [comment]
Yes, once the pane is cleared, I can repair it by double-clicking on one of my directories that belongs to an open project.

**Comment by Naatan** [comment]
Wait, how do you double click a directory if the Places pane is empty?

**Comment by brianjcohen** [comment]
The only thing that gets cleared is the part of the left-hand column that shows the directory/file structure and listings for the directory I'm currently in.  The next section down in that column, labeled "Projects", still shows my open projects and the folders I've added to those projects via "Add existing (remote) folder...".  Since those don't get cleared I can just re-doubleclick on any of those folders and the section above it will re-populate with the directory contents.

**Comment by Naatan** [comment]
I'm not sure what else it could be. The fact that it happens with a clean profile (safe-mode) is very odd given that I cannot reproduce it.

@mitchell-as can you try this on Edit? On the off chance that it is something Ubuntu specific.

**Comment by mitchell-as** [comment]
I cannot reproduce this one either. The pane does not go blank for me at all. If it's reloading once I click "Ok", then it is so fast I cannot see the refresh.

**Comment by brianjcohen** [comment]
What additional information (eg config files, debug output) could I provide to help you guys reproduce it?

**Comment by Naatan** [comment]
@brianjcohen given that you can reproduce it with a clean profile there isn't anything you could provide that would help us reproduce it. We need to go through you to find out what the issue is, unfortunately.

Could you make a short screen recording of the problem? Preferrably on a clean profile (safe mode). Perhaps that will shed some light.

**Comment by brianjcohen** [comment]
Here you go: https://dl.dropboxusercontent.com/u/315771/tmp/2016-06-28_Komodo-Edit-Places-Clearing-Bug.ogv

**Comment by Naatan** [comment]
Perfect! That showed us exactly what the problem is. Your project root directory is not the same as the directory that you appear to be pointing Places at. I guess you are adding an "existing folder" as a way to force Komodo to look at a folder that is not in the same place as your project file? You do not need to do this, the actual solution is much simpler: simply go into project preferences and modify you Project base directory to any directory that you prefer, eg. https://imgur.com/jnn2quf

The folders you can add through the project widget (bottom left of screen) are considered additional folders, not root folders. And by saving your project Komodo essentially reopens it, which would point it at your project root which appears to be empty?

**Comment by Naatan** [comment]
Hold on I spoke to soon, you were saving file preferences - not project preferences. While most of my response remains relevant it does seem like there is a bug here, which is that your Places root appears to change when changing file prefs on a file that is inside a project "existing" directory. Investigating..

**Comment by Naatan** [comment]
Reproduced - the steps in my previous response seem to trigger the Places root directory to change.

**Comment by tonyattwood** [comment]
Hello,

Just take  a  look at these  awesome creatures, I  love them! More info here <http://famous.pricelessmodels.com/e4sodmw>

Warmest, tony.attwood@ntlworld.com
